compiler/passes/normalize.cpp

```diff
@@ -2021,6 +2021,49 @@ static void normalizeCallToConstructor(CallExpr* call) {
   }
 }
 
+static IfExpr* getParentIfExpr(CallExpr* call) {
+  if (BlockStmt* parentBlock = toBlockStmt(call->parentExpr)) {
+    if (parentBlock->length() == 1) {
+      if (IfExpr* parentIf = toIfExpr(parentBlock->parentExpr)) {
+        if (parentIf->getThenStmt() == parentBlock ||
+            parentIf->getElseStmt() == parentBlock) {
+          return parentIf;
+        }
+      }
+    }
+  }
+
+  return NULL;
+}
+
+static bool callNeedsAnOwner(CallExpr* call) {
+  INT_ASSERT(call->isPrimitive(PRIM_NEW));
+
+  bool hasOwner = true;
+  if (IfExpr* parentIf = getParentIfExpr(call)) {
+    if (parentIf == parentIf->getStmtExpr()) {
+      hasOwner = false;
+    }
+  }
+  else {
+    if (call == call->getStmtExpr()) {
+      hasOwner = false;
+    }
+  }
+
+  if (hasOwner) return false;
+
+  if (call->numActuals() >= 1) {
+    if (SymExpr *typeSE = toSymExpr(call->get(1))) {
+      if (isDecoratedClassType(typeSE->symbol()->type)) {
```

Comment:
We can do this in the normalizer but it might not apply to `new t` where `t` is a type alias or a function returning a type

Reply:
Ah good point. Well... we can remove this check altogether and create temporaries for any `new t` as long as they are not assigned to a variable.